### PR TITLE
Display tests count change for every commit from parent commit(if parent have a tests)

### PR DESCRIPTION
**Pull request by shooter0106**

### CODE REVIEW

File: client/src/redux/sagas/commitsTestmanSaga.js [hunk 1]
```diff
@@ -6,8 +6,15 @@ import { setTestman, setTestmanError } from '../actions';
...
     const commits = yield select(state => state.commits);
-    const shas = commits.map(commit => commit.sha);
-    const testResults = yield call(fetchTests, shas[9], shas[0], 1);
+    const shas = [];
+    commits.forEach((commit, index) => {
+      if (index === 0) {
+        shas.push(commit.sha);
+      }
+
+      shas.push(commit.parents[0].sha);
```
Comment (Extravert-ir) [inline client/src/redux/sagas/commitsTestmanSaga.js]: What this change is intended for? I can't figure out, for me it just skips the second sha

File: client/src/components/Commits.js [hunk 2]
```diff
@@ -18,6 +19,7 @@ class Commits extends React.Component {
           {...commit}
           builds={this.props.build[commit.sha]}
           tests={this.props.testData[commit.sha]}
+          previousTests={this.props.testData[commit.parents[0].sha]}
```
Comment (Extravert-ir) [inline client/src/components/Commits.js]: Do we need a whole array of previous tests at all? Maybe we should make an object with {testType: testCount} data in it?

File: client/src/redux/sagas/commitsTestmanSaga.js (hunk 1 above)
Comment (shooter0106) [reply client/src/redux/sagas/commitsTestmanSaga.js]: In GitHub's API response every commit have a hash of it's parent commit. So, instead using own hashes, i build list of parents hashes and separately inserts first commit hash first. Further, i use this list in request to Testman's API.

File: client/src/components/Commits.js (hunk 2 above)
Comment (shooter0106) [reply client/src/components/Commits.js]: No, at current moment whole test objects is not required. I will change this behavior.

File: client/src/redux/sagas/commitsTestmanSaga.js (hunk 1 above)
Comment (Extravert-ir) [reply client/src/redux/sagas/commitsTestmanSaga.js]:
```
      let shas = [];
      if (commits.length > 0) {
        shas = [commits[0].sha, ...commits.map(commit => commit.parents[0].sha)];
      }
```

File: client/src/components/Commits.js [hunk 3]
```diff
@@ -103,6 +105,29 @@ const mapStateToProps = ({
...
+function extractCommitsParentTestsCount(testData, commit) {
+  let tests = testData[commit.parents[0].sha];
+
+  if (!tests) {
+    return {};
+  }
+
+  let result = {};
+
+  tests.forEach(test => {
```
Comment (Extravert-ir) [inline client/src/components/Commits.js]: Use `for...of` please

File: client/src/redux/sagas/commitsTestmanSaga.js (hunk 1 above)
Comment (Extravert-ir) [reply client/src/redux/sagas/commitsTestmanSaga.js]: I may be biased, but this look more straightforward to me :)

**Pull request merged**